### ISSUE TITLE
Ignore Index and Mean metrics

### ISSUE THREAD

**Issue by lebellig** [VMarsocci/pangaea-bench#115]
The current implementation of SegEvaluator does not take ignore_index into account when computing the mean metrics. For example, if the ignore_index removes a class, the mean metric is still obtained by dividing the metric by the number of classes instead of the number of classes - 1.
--> Check if the ignore index is in the range of class indices and then divide by the number of considered classes and not all classes.

**Comment by gastruc** [comment]
Yes, indeed, for example for PASTIS, it is a bit harsh to make the network try to guess the void label class

**Comment by lauraset** [comment]
Hi, @lebellig . It seems that `num_classes` means the number of valid classes, and is out of the range of class indices.
